### PR TITLE
Add `Externe ID` in PO-SO invoices

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/15086

### CODE REVIEW

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+-- Field: Invoice (Vendor)_OLD(183,D) -> Invoice(290,D) -> Externe ID
+-- Column: C_Invoice.ExternalId
+-- 2023-04-06T21:09:05.735Z
+UPDATE AD_Field SET ReadOnlyLogic='',Updated=TO_TIMESTAMP('2023-04-06 22:09:05','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=713823
+;
+
+-- Field: Invoice (Vendor)_OLD(183,D) -> Invoice(290,D) -> Externe ID
+-- Column: C_Invoice.ExternalId
+-- 2023-04-06T21:13:51.918Z
+UPDATE AD_Field SET IsAlwaysUpdateable='', ReadOnlyLogic='@ExternalId@!NULL',Updated=TO_TIMESTAMP('2023-04-06 22:13:51','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=713823
```
Comment (adi-stefan) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql:67]: We do not want this logic in the public repo. The field shall be editable only when the record is not yet processed.

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+;
+
+-- 2023-04-06T22:08:07.193Z
+/* DDL */ select AD_Element_Link_Create_Missing_Field(713824)
+;
+
+-- UI Element: Invoice (Customer)_OLD(167,D) -> Invoice(263,D) -> main -> 20 -> Externe ID
+-- Column: C_Invoice.ExternalId
+-- 2023-04-06T22:11:37.741Z
+INSERT INTO AD_UI_Element (AD_Client_ID,AD_Field_ID,AD_Org_ID,AD_Tab_ID,AD_UI_ElementGroup_ID,AD_UI_Element_ID,AD_UI_ElementType,Created,CreatedBy,IsActive,IsAdvancedField,IsAllowFiltering,IsDisplayed,IsDisplayedGrid,IsDisplayed_SideList,IsMultiLine,MultiLine_LinesCount,Name,SeqNo,SeqNoGrid,SeqNo_SideList,Updated,UpdatedBy) VALUES (0,713824,0,263,540027,616644,'F',TO_TIMESTAMP('2023-04-06 23:11:31','YYYY-MM-DD HH24:MI:SS'),100,'Y','N','N','Y','N','N','N',0,'JobRouter Instance ID',32,0,0,TO_TIMESTAMP('2023-04-06 23:11:31','YYYY-MM-DD HH24:MI:SS'),100)
```
Comment (adi-stefan) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql:108]: The UI Element shall be named `Externe ID`, not `JobRouter Instance ID`.

**Review by adi-stefan — CHANGES_REQUESTED** [review]
see my comments above

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+;
+
+-- 2023-04-06T20:51:11.586Z
+/* DDL */ select AD_Element_Link_Create_Missing_Field(713823)
+;
+
+-- UI Element: Invoice (Vendor)_OLD(183,D) -> Invoice(290,D) -> main -> 20 -> Externe ID
+-- Column: C_Invoice.ExternalId
+-- 2023-04-06T20:53:16.849Z
+INSERT INTO AD_UI_Element (AD_Client_ID,AD_Field_ID,AD_Org_ID,AD_Tab_ID,AD_UI_ElementGroup_ID,AD_UI_Element_ID,AD_UI_ElementType,Created,CreatedBy,IsActive,IsAdvancedField,IsAllowFiltering,IsDisplayed,IsDisplayedGrid,IsDisplayed_SideList,IsMultiLine,MultiLine_LinesCount,Name,SeqNo,SeqNoGrid,SeqNo_SideList,Updated,UpdatedBy,WidgetSize) VALUES (0,713823,0,290,540226,616643,'F',TO_TIMESTAMP('2023-04-06 21:53:15','YYYY-MM-DD HH24:MI:SS'),100,'Y','N','N','Y','N','N','N',0,'Externe ID',50,0,0,TO_TIMESTAMP('2023-04-06 21:53:15','YYYY-MM-DD HH24:MI:SS'),100,'S')
```
Comment (adi-stefan) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql:38]: Also consider moving to advanced view. Most of our users are not interested in this field, hence we do not have it displayed at all.

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql:108]: As above, consider moving the field to advanced view.

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+-- Field: Invoice (Vendor)_OLD(183,D) -> Invoice(290,D) -> Externe ID
+-- Column: C_Invoice.ExternalId
+-- 2023-04-06T21:01:04.970Z
+UPDATE AD_Field SET DisplayLogic='@ExternalId@!NULL',Updated=TO_TIMESTAMP('2023-04-06 22:01:04','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=713823
+;
+
+-- Column: C_Invoice.ExternalId
+-- 2023-04-06T21:04:08.565Z
+UPDATE AD_Column SET IsAlwaysUpdateable='Y',Updated=TO_TIMESTAMP('2023-04-06 22:04:08','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Column_ID=572375
+;
```
Comment (adi-stefan) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql:50]: This should also go.

**Review by adi-stefan — APPROVED** [review]

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql (hunk 4 above)
Comment (riadmen) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5684510_sys_gh15086_JobRouter_Instance_ID.sql]: done

**Pull request merged**